src/Hashids.net/Hashids.cs

```diff
@@ -207,6 +207,80 @@ public string EncodeLong(long number)
         /// <returns>Array of 64-bit integers.</returns>
         public long[] DecodeLong(string hash) => GetNumbersFrom(hash);
 
+        /// <summary>
+        /// Decodes the provided hash into numbers.
+        /// </summary>
+        /// <param name="hash">Hash string to decode.</param>
+        /// <returns>A 64-bit integer.</returns>
+        /// <exception cref="T:HashidsNet.MultipleResultsException">If the decoded hash returns more than one long</exception>
+        public long DecodeSingleLong(string hash)
+        {
+            var numbers = GetNumbersFrom(hash);
+
+            if (numbers.Length == 1) return numbers[0];
+            else throw new MultipleResultsException("The hash provided yielded more than one result.");
```

Comment:
This will throw even if numbers.Length == 0. And text of the exception will be incorrect.

Reply:
True, GetNumberFrom can indeed return Array.Empty, let me take care of that.